src/biogeochem/SPMMod.F90

```diff
@@ -0,0 +1,804 @@
+module SPMMod
+
+!-----------------------------------------------------------------------
+!BOP
+!
+! !MODULE: SPMMod
+!
+! !DESCRIPTION:
+! Sparse matrix multiplication add addition
+!
+! Author: Xingjie Lu
+!
+!EOP
+!-----------------------------------------------------------------------
+
+  use shr_kind_mod                   , only : r8 => shr_kind_r8
+  use decompMod                      , only : bounds_type
+  implicit none
+  private
+
+  type, public :: sparse_matrix_type
+ 
+  !sparse matrix is in COO format, Both row index and column index should be in ascending order. 
+  !Row index should change faster than Column index to ensure SPMP_AB work properly.
+
+     real(r8), pointer :: M(:,:)
+     integer , pointer :: RI(:) !Row index
+     integer , pointer :: CI(:) !Column index
+     integer NE
+     integer SM
+     integer num_unit
+     integer begu
+     integer endu
+  
+  contains
+    
+    procedure, public :: InitSM
```

Comment:
Give some comments to describe what these are.

Reply:
still needs to be done.

Reply:
done